azure-slurm/test/slurmcc_test/topology_test.py

```diff
@@ -300,39 +306,66 @@ def test_generate_topo_file():
         actual= file.read()
     assert result==actual
 
-def test_write_slurm_topology():
+def test_write_tree_topology():
     """
-    Test the write_slurm_topology method of the Topology class.
+    Test the write_tree_topology method of the Topology class.
 
     This test performs the following steps:
     1. Creates an instance of the Topology class with a specified output file.
     2. Sets the topo_file attribute to a predefined input topology file.
     3. Groups device GUIDs per switch using the group_guids_per_switch method.
     4. Identifies torsets using the identify_torsets method.
     5. Groups hosts by torset using the group_hosts_by_torset method.
-    6. Writes the slurm topology to the output file using the write_slurm_topology method.
+    6. Writes the slurm topology to the output file using the write_tree_topology method.
     7. Reads the generated output file and compares it with an expected output file.
     
     The test asserts that the content of the generated output file matches the expected output file.
     """
     slutil.srun=run_parallel_cmd
     slutil.run=run_command
     output= 'test/slurmcc_test/topology_test_output/slurm_topology_1.txt'
-    test_obj   = Topology("hpc",output,TESTDIR)
+    test_obj   = Topology("hpc",output,"fabric","tree",TESTDIR)
     test_obj.get_hostnames()
     test_obj.retrieve_guids()
     test_obj.topo_file = 'test/slurmcc_test/topology_test_input/topology.txt'
     test_obj.device_guids_per_switch =  test_obj.group_guids_per_switch()
     test_obj.host_to_torset_map = test_obj.identify_torsets()
     test_obj.torsets = test_obj.group_hosts_by_torset()
-    test_obj.write_slurm_topology()
+    test_obj.write_tree_topology()
```

Comment:
Another reason to split the `write_tree|block_topology()` methods into a 
`topo_str = generate_tree|block_topology()
do_something_with_topology(topo_str)
`
then you could test the generate method with many inputs trivially by just asserting against the "topo_str" output without a ton of golden input/output files.